public/app/core/components/AppChrome/AppChromeService.tsx

```diff
@@ -58,9 +65,15 @@ export class AppChromeService {
   }
 
   public update(update: Partial<AppChromeState>) {
+    const returnToPrevious = getReturnToPrevious();
     const current = this.state.getValue();
     const newState: AppChromeState = {
       ...current,
+      returnToPrevious: {
+        ...current.returnToPrevious,
```

Comment:
Does `current.returnToPrevious` have other properties apart from `href` and `title`? If not, I wonder if we need the original `...current.returnToPrevious` here, since all its properties are overwritten anyways.